### ISSUE TITLE
Q about impact on the Campaign

### ISSUE THREAD

**Issue by Howling-Dog** [BattletechModders/JK_Rarity#1]
Given that during the campaign you will be doing a lot of missions against the Aurigan Directorate, most of your foes will have the tag `"unit_common_AuriganDirectorate"`, correct?  This will lead to seeing a huge number of say Urbanmechs, and none of certain others.

Seems like it might make things somewhat repetitive, or am I missing something?

**Comment by JustinKaseToo** [comment]
Yes, it can lead to seeing some 'Mechs/variants more often than others and other variants hardly at all if you don't fight that faction.  This is controlled via the rarity levels of each faction tag AND the lance role.

If a lance calls for a Light Tank 'Mech, you may get the Urbie for Directorate.  But if it calls for a Light Jumping Scout, you will not.

The overall effect is a lot more useful when playing with something where you have more than just the Vanilla 'Mechs - but it still has impact even on just Vanilla in that you will see more Urbies for Liao, more Panthers for Kurita, more D variants for Davion, etc.

**Comment by Howling-Dog** [comment]
Follow up in a similar vein, I cloned your repo and spent some time looking through the `lancedef` files to see how spawns were chosen.  It appears that the only time extremely rare mechs will spawn is from the `dynamic_solo` missions.  Does seem to be possible to get very rare mechs from the third mech in a four mech lance but not on the first, second or fourth mech.  That sound correct?

**Comment by JustinKaseToo** [comment]
For the most part, the Ext_Rare are limited to the solo's to help limit them further.  It was also intended to be able to be used for Flashpoints or homemade lancedefs if you wanted to specifically call a rare 'Mech.

I am waiting to see what they do with things in 1.6, as it looks like there is a trend toward converting rarity / tables into CSV files, so I am hoping this is something that HBS will be adding in their own take on Rarity.

IF not, I'll be doing a bit of a revamp and review of it and adding into the chances of rare items more often.

**Comment by Howling-Dog** [comment]
TY
